### PR TITLE
Change Hashing Function from Blake2b to Sha3

### PR DESCRIPTION
**Pull request by terenc3t**
Switched from blake2b to sha3/keccak-256 for our `Hash` function

See rationale: https://github.com/ethereum/eth2.0-specs/pull/227

### CODE REVIEW

File: beacon-chain/chaintest/backend/simulated_backend_test.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ func TestRunShuffleTest(t *testing.T) {
 	}
 	testCase := &ShuffleTestCase{
 		Input:  []uint32{1, 2, 3, 4, 5},
-		Output: []uint32{2, 4, 1, 3, 5},
+		Output: []uint32{2, 4, 5, 1, 3},
```
Comment (nisdas) [inline beacon-chain/chaintest/backend/simulated_backend_test.go]: why was this changed ?

File: shared/hashutil/hash.go [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 
 // Hash defines a function that returns the
-// blake2b hash of the data passed in.
+// Keccak-256/SHA3 hash of the data passed in.
 // https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#appendix
 func Hash(data []byte) [32]byte {
 	var hash [32]byte
-	h := blake2b.Sum512(data)
+
+	h := sha3.Sum256(data)
```
Comment (nisdas) [inline shared/hashutil/hash.go]: which sha3 is this ? the keccak-256 used by ethereum is different then what is finalised as sha3 as a standard 
For reference: 
https://ethereum.stackexchange.com/questions/550/which-cryptographic-hash-function-does-ethereum-use

File: beacon-chain/chaintest/backend/simulated_backend_test.go (hunk 1 above)
Comment (terenc3t) [reply beacon-chain/chaintest/backend/simulated_backend_test.go]: Shuffle test uses the hashing function. Diff hashing algorithm produces different shuffled output...

File: shared/hashutil/hash.go (hunk 2 above)
Comment (rauljordan) [reply shared/hashutil/hash.go]: Yes this is different @terenc3t - I think we should just use go-ethereum/common with its BytesToHash methods everywhere instead and deprecate our hashutils package
